test/models/hymod/include/Et_All_Methods_Test.cpp

```diff
@@ -0,0 +1,1032 @@
+
+class EtCalcKernelTest : public ::testing::Test {
+
+    protected:
+
+    EtCalcKernelTest() {
+
+    }
+
+    ~EtCalcKernelTest() override {
+
+    }
+
+    void SetUp() override;
+
+    void TearDown() override;
+
+    void setupArbitraryExampleCase();
+
+};
+
+void EtCalcKernelTest::SetUp() {
+    setupArbitraryExampleCase();
+}
+
+void EtCalcKernelTest::TearDown() {
+
+}
+
+void EtCalcKernelTest::setupArbitraryExampleCase() {
+
+}
+
+
+TEST_F(EtCalcKernelTest, TestEnergyBalanceMethod)
+{
+  struct evapotranspiration_options set_et_options;
+  double et_m_per_s;
+  int et_method_option;
+
+
+  // FLAGS
+  int yes_aorc; // if TRUE then using AORC forcing data- if FALSE then we must calculate incoming short/longwave rad.
+  int yes_wrf;  // if TRUE then we get radiation winds etc. from WRF output.  TODO not implemented.
+
+  struct aorc_forcing_data aorc;
+
+  struct evapotranspiration_options et_options;
+  struct evapotranspiration_params  et_params;
+  struct evapotranspiration_forcing et_forcing;
+  struct intermediate_vars inter_vars;
+
+  struct surface_radiation_params   surf_rad_params;
+  struct surface_radiation_forcing  surf_rad_forcing;
+
+  struct solar_radiation_options    solar_options; 
+  struct solar_radiation_parameters solar_params;
+  struct solar_radiation_forcing    solar_forcing;
+  struct solar_radiation_results    solar_results;
+
+
+  double wind_speed_at_2_m;
+  double et_mm_per_d;
+  double numerator,denominator;
+  double saturation_vapor_pressure_Pa;
+  double actual_vapor_pressure_Pa;
+
+  //###################################################################################################
+  // THE VALUE OF THESE FLAGS DETERMINE HOW THIS CODE BEHAVES.  CYCLE THROUGH THESE FOR THE UNIT TEST.
+  //###################################################################################################
+  // Set this flag to TRUE if meteorological inputs come from AORC
+  set_et_options.yes_aorc = TRUE;                      // if TRUE, it means that we are using AORC data.
+  set_et_options.shortwave_radiation_provided = FALSE;
+  et_options.yes_aorc = set_et_options.yes_aorc;
+  et_options.shortwave_radiation_provided = set_et_options.shortwave_radiation_provided;
+
+
+  // -----UNIT TEST RESULTS:-----
+
+  // energy balance method:
+  // calculated instantaneous potential evapotranspiration (PET) =8.594743e-08 m/s
+  // calculated instantaneous potential evapotranspiration (PET) =7.425858 mm/d
+
+
+  //###################################################################################################
+  // MAKE UP SOME TYPICAL AORC DATA.  THESE VALUES DRIVE THE UNIT TESTS.
+  //###################################################################################################
+  //read_aorc_data().  TODO: These data come from some aorc reading/parsing function.
+  //---------------------------------------------------------------------------------------------------------------
+
```

Comment:
These structs that are used across multiple tests should be created in the `setup` and reused for each test.  This will simplify the test code and make it easier to understand what is being tested.